# HG changeset patch
# User Andreas Weber <octave@josoansi.de>
# Date 1546552087 -3600
#      Thu Jan 03 22:48:07 2019 +0100
# Branch stable
# Node ID c67501d555418957ad25e5a8a0daa64e552bdbb6
# Parent  25dfa8c96b7b2f241f07cb53dc14277c1fed2d95
ls-mat-ascii.cc: Fix static analyzer detected issues (bug #55347)

ls-mat-ascii.cc: also use is.fail () to exit loop

diff --git a/libinterp/corefcn/ls-mat-ascii.cc b/libinterp/corefcn/ls-mat-ascii.cc
--- a/libinterp/corefcn/ls-mat-ascii.cc
+++ b/libinterp/corefcn/ls-mat-ascii.cc
@@ -91,17 +91,17 @@ get_mat_data_input_line (std::istream& i
             {
               if (! have_data && c != ' ' && c != '\t')
                 have_data = true;
 
               retval += c;
             }
         }
     }
-  while (! (have_data || is.eof ()));
+  while (! (have_data || is.eof () || is.fail ()));
 
   return retval;
 }
 
 static void
 get_lines_and_columns (std::istream& is,
                        octave_idx_type& nr, octave_idx_type& nc,
                        const std::string& filename = "",
